### PR TITLE
[Validator] Improve type for the `mode` property of the `Bic` constraint

### PR DESCRIPTION
**Pull request by MrYamous**
Following [58810](https://github.com/symfony/symfony/pull/58810) in code

### CODE REVIEW

File: reference/constraints/Bic.rst [hunk 1]
```diff
@@ -124,18 +124,13 @@ Parameter        Description
...
-* ``strict`` validates the given value without any modification;
-* ``case-insensitive`` converts the given value to uppercase before validating it.
-
-.. tip::
-
-    The possible values of this option are also defined as PHP constants of
-    :class:`Symfony\\Component\\Validator\\Constraints\\BIC`
-    (e.g. ``BIC::VALIDATION_MODE_CASE_INSENSITIVE``).
+* ``BIC::VALIDATION_MODE_STRICT`` validates the given value without any modification;
+* ``BIC::VALIDATION_MODE_CASE_INSENSITIVE`` converts the given value to uppercase before validating it.
```
Comment (OskarStark) [inline reference/constraints/Bic.rst:133]:
```suggestion
* ``Bic::VALIDATION_MODE_STRICT`` validates the given value without any modification;
* ``Bic::VALIDATION_MODE_CASE_INSENSITIVE`` converts the given value to uppercase before validating it.
```

File: reference/constraints/Bic.rst [hunk 2]
```diff
@@ -124,18 +124,13 @@ Parameter        Description
 ``mode``
 ~~~~~~~~
 
-**type**: ``string`` **default**: ``strict``
+**type**: ``string`` **default**: ``BIC::VALIDATION_MODE_STRICT``
 
-This option defines how the BIC is validated:
+This option defines how the BIC is validated. Available constants are defined
+in the :class:`Symfony\\Component\\Validator\\Constraints\\BIC` class.
```
Comment (OskarStark) [inline reference/constraints/Bic.rst:130]:
```suggestion
in the :class:`Symfony\\Component\\Validator\\Constraints\\Bic` class.
```

File: reference/constraints/Bic.rst [hunk 3]
```diff
@@ -124,18 +124,13 @@ Parameter        Description
 ``mode``
 ~~~~~~~~
 
-**type**: ``string`` **default**: ``strict``
+**type**: ``string`` **default**: ``BIC::VALIDATION_MODE_STRICT``
```
Comment (OskarStark) [inline reference/constraints/Bic.rst:127]:
```suggestion
**type**: ``string`` **default**: ``Bic::VALIDATION_MODE_STRICT``
```

**Comment by javiereguiluz** [comment]
Thanks Matthieu!

Oskar, I made your requested changes while merging. Thanks!

**Pull request merged**